Dapper/DynamicParameters.cs

```diff
@@ -463,11 +463,13 @@ public DynamicParameters Output<T>(T target, Expression<Func<T, object>> express
 
                 if (parameters.TryGetValue(dynamicParamName, out parameter))
                 {
-                    parameter.ParameterDirection = parameter.AttachedParam.Direction = ParameterDirection.InputOutput;
+                    parameter.ParameterDirection = ParameterDirection.InputOutput;
+                    parameter.Size = sizeToSet;

```

Comment:
I had to split both the direction and size assignments into two due to `parameter.AttachedParam` being null if a template object is not passed to the `DynamicParameters` constructor.

Happy to not change the `Size` property at all here, and only do it if `AttachedParam` is not null.

Obviously `Direction` will still need to change.